File: OOPRecords/Student.cs
using System;

namespace OOPRecords
{
    public class Student
    {
        public virtual int Id { get; set; }

        public virtual string FirstName { get; set; }

        public virtual string LastName { get; set; }

        public virtual DateTime DateOfBirth { get; set; }


        public Student(int id, string firstName, string lastName, DateTime dateOfBirth)
        {
            Id = id;
            FirstName = firstName;
            LastName = lastName;
            DateOfBirth = dateOfBirth;
        }


        public int Age()
        {
            var today = DateTime.Today;
            int age = today.Year - DateOfBirth.Year;

            if (today.Month < DateOfBirth.Month || (today.Month == DateOfBirth.Month && today.Day < DateOfBirth.Day))
                age--;
            return age;
        }


        public override string ToString()
        {
            return $"{FirstName} {LastName}, Age {Age()}";
        }
    }
}

File: ModelVB/Student.vb
Public Class Student
    Public Overridable Property Id As Integer
    Public Overridable Property FirstName As String
    Public Overridable Property LastName As String
    Public Overridable Property DateOfBirth As Date
    Public Function ValidateDateOfBirth(ByVal dob As DateTime) As String
        Return If(dob > DateTime.Today, "Date of Birth cannot be after today", Nothing)
    End Function

    Public Overridable Property Tutor As Teacher
    Public Overridable Property StudentNumber As String

    Public Function Age() As Integer
        Dim today = Date.Today
        Dim ageToday As Integer = today.Year - DateOfBirth.Year
        If today.Month < DateOfBirth.Month OrElse (today.Month = DateOfBirth.Month AndAlso today.Day < DateOfBirth.Day) Then ageToday -= 1
        Return ageToday
    End Function

    Public Overrides Function ToString() As String
        Return $"{FirstName} {LastName}, Age {Age()}"
    End Function
End Class

